src/field/legend_cutscene_camera.c

```diff
@@ -0,0 +1,987 @@
+    BIRD_FINAL_APPROACH_TASK_STATE_FLAP_DOWN,
+    BIRD_FINAL_APPROACH_TASK_STATE_DELAY__ALT_ANIM,
+    BIRD_FINAL_APPROACH_TASK_STATE_DELAY__CRY,
+    BIRD_FINAL_APPROACH_TASK_STATE_EXPANDING_CIRCLE_EFFECT,
+    BIRD_FINAL_APPROACH_TASK_STATE_WAIT_AFTER_ANIM,
+    BIRD_FINAL_APPROACH_TASK_STATE_FINISH,
+} BirdFinalApproachTaskState;
+
+typedef struct ClearBellCutsceneCamera {
+    Field3dObjectTaskManager *field3dObjectTaskManager;
+    Field3dObjectTask *draw3dTask;
+    GFCameraTranslationWrapper *translation;
+    VecFx32 lookAtTarget;
+    VecFx32 lookAtPos;
+    int gameVersion;
+} ClearBellCutsceneCamera;  // size: 0x28
+
+typedef struct LugiaArrivesCutsceneCamera {
+    fx32 distance;
+    CameraAngle angle;
+    VecFx32 targetDistanceVec;
+    u16 perspectiveAngle;
+    u16 gameVersion;
+    fx32 clippingPlaneNear;
+    fx32 clippingPlaneFar;
+    GFCameraTranslationWrapper *translation;
+} LugiaArrivesCutsceneCamera;
+
+typedef struct ClearBellCutscene3dObjectTaskData_SoulSilver {
+    Field3dModel model;
+    Field3DModelAnimation anims[2];
+    Field3dObject object;
+} ClearBellCutscene3dObjectTaskData_SoulSilver;
+
+typedef struct ClearBellCutscene3dObjectTaskData_HeartGold {
+    NNSG2dScreenData *bg2ScrnData[6];
+    NNSG2dScreenData *bg3ScrnData[6];
+    void *bg2ScrnRaw[6];
+    void *bg3ScrnRaw[6];
+    u8 bg2X;
+    u8 bg3X;
+    u8 bgY;
+    u8 switchTilemapDelayTimer;
+    BgConfig *bgConfig;
+    int tilemapIndex;
+} ClearBellCutscene3dObjectTaskData_HeartGold;
+
+typedef struct ClearBellCutscene3dObjectTaskData {
+    NNSFndAllocator allocator;
+    u8 state;
+    u8 shimmerClearBellOnly;
+    u16 gameVersion;
+    Field3dModel clearBellRisingModel;
+    Field3dModel clearBellShimmerModel;
+    Field3dModel cornerBellModel;
+    Field3DModelAnimation clearbellRisingAnims[3];
+    Field3DModelAnimation clearBellShimmerAnims[2];
+    Field3DModelAnimation cornerBellAnims[1];
+    Field3dObject clearBellRisingObject;
+    Field3dObject clearBellObject;
+    Field3dObject cornerBellObjects[4];
+    ClearBellCutscene3dObjectTaskData_SoulSilver wavesEffect;
+    ClearBellCutscene3dObjectTaskData_HeartGold leavesEffect;
+    SysTask *task;
+    u32 birdModelNum;
+    const CameraParam *cameraParam;
+    const VecFx32 *cameraOffset;
+} ClearBellCutscene3dObjectTaskData;  // size: 0x4B8
+
+typedef struct LegendCutsceneLugiaEyeGlimmerTaskData {
+    NNSFndAllocator allocator;
+    Field3dModel model;
+    Field3DModelAnimation anims[3];
+    Field3dObject object;
+    int eyeGlimmerDelayTimer;
+} LegendCutsceneLugiaEyeGlimmerTaskData;
+
+typedef struct BirdFinalApproachTaskData {
+    fx32 yDelta;
+    fx32 yEnd;
+    u16 delayTimer;
+    u16 animTimer;
+    NNSFndAllocator allocator;
+    Field3dModel model;
+    Field3DModelAnimation anims[3];
+    Field3dObject object;
+    u16 gameVersion;
+    u16 betweenFlapsDelayTimer;
+} BirdFinalApproachTaskData;
+
+static void startBellShimmer(FieldSystem *fieldSystem);
```

Comment:
Match naming style to have capital letters at the start of functions (even if they're static)